### DISCUSSION TITLE
Improved Data Tab UI

### DISCUSSION THREAD

**Discussion by vijayprasanna13** [hasura/graphql-engine#9846] [General]
Hey folks, we've released a new version of the Data Tab UI for the console that offers a better experience for datasets with large schemas.  

![Screenshot 2023-08-23 at 7 36 59 PM](https://github.com/hasura/graphql-engine/assets/11921040/1d8d8a33-7d3c-4bd0-976d-a35ed965ded6)

### Updates
1. Less Introspection to your Databases and no more `SQL timeout errors` for large datasets.
2. The Navigation Tree can handle 10k+ tables and offers complete text-search across all your connected databases.
3. The Tracking section has been unified under the Database level and you can see all your trackable items - tables, foreign keys, functions from the same places without having to switching between schemas.
4. New UI for Browse Rows, Modify and Permissions Tab.

### How to Enable
You can try out the early version of this feature starting from `v2.33.0-beta.1` onwards by enabling the feature flag from the `Settings` tab. 
![feature-flag](https://github.com/hasura/graphql-engine/assets/11921040/e3a6982d-4b83-44e9-ad84-e8ce9092303a)

**Please note that this is an experimental feature and is missing some functionality like DDL support, Insert/Modify row etc.** We'll continue to add more updates to UI and we are happy to your hear your early feedback!

**Comment by anish-pr** [comment]
@vijayprasanna13  This seems to have brought improvements in the UI performance, I tried a database with almost 1000+ tables (including partitions)
I noticed the edit options were not there, but the overall UI getting hung up issue is no longer happening in this build.
When can we expect a full-fledged stable version for this?

**Reply by vijayprasanna13** [reply]
Thanks @anish-pr for trying it out! Happy to hear that it's able to handle 1000+ tables. Yes, the UI is missing the DDL features, insert/modify tab. We're working on adding them as soon as possible and we'll continue to roll them out in the subsequent releases, next in line being the Insert/Modify Row functionality and Apollo federation support. 
Are there priority features that you'd like to have first? We can definitely move them up our list.

**Reply by anish-pr** [reply]
Insert/ modify ddl and records with this new ui would really be helpful @vijayprasanna13  please do let me know once it is available. It does hang bit when i add relationships

**Reply by noisyscanner** [reply]
+1 for support to insert/modify rows @vijayprasanna13 
Do you have an update on how this feature is progressing and when we could expect it to be done?
Cheers
